include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -108,23 +118,216 @@ __pattern_transform_reduce(_ExecutionPolicy&& __exec, _ForwardIterator __first,
     return __res;
 }
 
+template< bool _Inclusive>
```

Comment:
What's the purpose for it being a class?

Reply:
There used to be more shared code in the methods, but I suppose at this point they can be free-standing functions.

Reply:
There is more of an argument for it being a class now that I removed the `__i_kernel_name` usage and rely on alias templates, which can only be defined in class scope or namespace scope.